ProcessLib/UncoupledProcessesTimeLoop.cpp

```diff
@@ -330,16 +335,37 @@ std::unique_ptr<UncoupledProcessesTimeLoop> createUncoupledProcessesTimeLoop(
         //! \ogs_file_param{prj__time_loop__global_process_coupling}
         = config.getConfigSubtreeOptional("global_process_coupling");
 
-    std::unique_ptr<NumLib::ConvergenceCriterion> coupling_conv_crit = nullptr;
+    std::vector<std::unique_ptr<NumLib::ConvergenceCriterion>>
+        global_coupling_conv_criteria;
     unsigned max_coupling_iterations = 1;
     if (coupling_config)
     {
         max_coupling_iterations
             //! \ogs_file_param{prj__time_loop__global_process_coupling__max_iter}
             = coupling_config->getConfigParameter<unsigned>("max_iter");
-        coupling_conv_crit = NumLib::createConvergenceCriterion(
-            //! \ogs_file_param{prj__time_loop__global_process_coupling__convergence_criterion}
-            coupling_config->getConfigSubtree("convergence_criterion"));
+
+        auto const& coupling_convergence_criteria_config =
+            //! \ogs_file_param{prj__time_loop__global_process_coupling__convergence_criteria}
+            coupling_config->getConfigSubtree("convergence_criteria");
+
+        std::vector<int> criterion_ids;
+        for (
+            auto coupling_convergence_criterion_config :
+            //! \ogs_file_param{prj__time_loop__global_process_coupling__convergence_criteria__convergence_criterion}
+            coupling_convergence_criteria_config.getConfigSubtreeList(
+                "convergence_criterion"))
+        {
+            //! \ogs_file_attr{prj__time_loop__global_process_coupling__convergence_criteria__convergence_criterion__id}
+            auto const id =
+                coupling_convergence_criterion_config.getConfigAttribute<int>(
+                    "id");
+            criterion_ids.push_back(id);
+
+            global_coupling_conv_criteria.push_back(
+                NumLib::createConvergenceCriterion(
+                    coupling_convergence_criterion_config));
+        }
+        BaseLib::reorderVector(global_coupling_conv_criteria, criterion_ids);
```

Comment:
Do the `criterion_ids` have any additional meaning, or are they just meant to put the `global_coupling_conv_criteria` into the right order?

Reply:
Is there a check that the number of convergence criteria == the number of processes?
Maybe it's better to drop the `criterion_ids` and take the `global_coupling_conv_criteria` in the order  they are read.

Reply:
Added an assertion for `global_coupling_conv_criteria` and dropped `id`.